DatadogInternal/Sources/Telemetry/Telemetry.swift

```diff
@@ -75,6 +75,67 @@ public protocol Telemetry {
     func send(telemetry: TelemetryMessage)
 }
 
+public extension Telemetry {
+    /// Starts a method call.
+    ///
+    /// - Parameters:
+    ///   - operationName: Platform agnostic name of the operation.
+    ///   - callerClass: The name of the class that calls the method.
+    ///   - samplingRate: The sampling rate of the method call. Value between `0.0` and `100.0`, where `0.0` means NO event will be processed and `100.0` means ALL events will be processed. Note that this value is multiplicated by telemetry sampling (by default 20%) and metric events sampling (hardcoded to 15%). Making it effectively 3% sampling rate for sending events, when this value is set to `100`.
+    ///
+    /// - Returns: A `MethodCalledTrace` instance to be used to stop the method call and measure it's execution time. It can be `nil` if the method call is not sampled.
+    func startMethodCalled(
```

Comment:
identifying method with just name and class seems naive but probably okay for current requirements.

For example `RequestBuilder` protocol we have which has same impl name everywhere.

There is concept of fully qualified name which gives the unique name of the method in the current context.

Reply:
Thanks! Some good points there

`operation_name` is agnostic of the platform because we can measure the same "behaviour" in both platforms, but they might have totally different function signature. For the class name we currently do this:
```
static let className = { String(reflecting: Recorder.self) }()
```

Which produces: `"DatadogSessionReplay.Recorder"`

`Method Call` is indeed first and simplistic example and if we need we can expand this model for the future use cases.